### PR TITLE
New per-user and per-workflow endpoints

### LINKED ISSUE #920: Expose per-user and per-workflow GET-job endpoints?
I guess the most reasonable URLs are

- [x]  `GET /api/v1/project/7/workflow/3/job/` -> all jobs of a given workflow (TODO)
- [x]  `GET /api/v1/project/7/job/3` -> one job of the current user
- [x]  `GET /api/v1/project/7/job/` -> all jobs of a given workflow
- [x]  `GET /api/v1/project/job/` -> all jobs of the current user (TODO)

### PR DESCRIPTION
**Pull request by ychiucco**
closes #920 

## Checklist before merging
- [x] I added an appropriate entry to `CHANGELOG.md`

### CODE REVIEW

File: tests/test_job_api.py [hunk 1]
```diff
@@ -206,3 +208,70 @@ async def test_get_job_list(
...
+                output_dataset_id=dataset.id,
+                workflow_id=workflow.id,
+            )
+
+        project2 = await project_factory(user)
+        workflow2 = await workflow_factory(project_id=project.id)
+        await workflow2.insert_task(task_id=task.id, db=db)
+        dataset2 = await dataset_factory(project_id=project.id)
+
+        await db.refresh(project)
```
Comment (ychiucco) [inline tests/test_job_api.py:259]: Why do we need this line?

File: tests/test_job_api.py [hunk 2]
```diff
@@ -206,3 +208,70 @@ async def test_get_job_list(
...
+        for _ in range(2):
+            await job_factory(
+                working_dir=tmp_path.as_posix(),
+                project_id=project2.id,
+                input_dataset_id=dataset2.id,
+                output_dataset_id=dataset2.id,
+                workflow_id=workflow2.id,
+            )
+
+        await db.refresh(project2)
```
Comment (ychiucco) [inline tests/test_job_api.py:269]: Why do we need this line?

Comment (tcompa) [reply tests/test_job_api.py]: Fixed by changing the fixture

File: tests/test_job_api.py (hunk 1 above)
Comment (tcompa) [reply tests/test_job_api.py]: Fixed by changing the `job_factory` fixture

File: fractal_server/app/api/v1/job.py [hunk 3]
```diff
@@ -22,11 +22,50 @@
...
+@router.get(
+    "/project/{project_id}/workflow/{workflow_id}/job/",
+    response_model=list[ApplyWorkflowRead],
+)
+async def get_workflow_jobs(
+    project_id: int,
+    workflow_id: int,
+    user: User = Depends(current_active_user),
+    db: AsyncSession = Depends(get_db),
+) -> list[ApplyWorkflowRead]:
```
Comment (tcompa) [inline fractal_server/app/api/v1/job.py:55]: This should be optional

**Review by tcompa — APPROVED** [review]

File: fractal_server/app/api/v1/job.py (hunk 3 above)
Comment (tcompa) [reply fractal_server/app/api/v1/job.py]: Now fixed, thanks!

**Pull request merged**